From 1997a599686e172deab806910271c4270ec99a7a Mon Sep 17 00:00:00 2001
From: Chen Liangjun <b36089@freescale.com>
Date: Wed, 31 Oct 2012 12:52:48 +0800
Subject: [PATCH 2226/2463] ENGR00231773-9 ASRC: increace convert speed while
 in ideal mode

Increase convert speed while ASRC is working on ideal ratio mode to
satisfy asrc plugin's timing request.

Signed-off-by: Chen Liangjun <b36089@freescale.com>
---
 drivers/mxc/asrc/mxc_asrc.c |    9 +++++++++
 1 files changed, 9 insertions(+), 0 deletions(-)

diff --git a/drivers/mxc/asrc/mxc_asrc.c b/drivers/mxc/asrc/mxc_asrc.c
index d16d716..b728fe1 100644
--- a/drivers/mxc/asrc/mxc_asrc.c
+++ b/drivers/mxc/asrc/mxc_asrc.c
@@ -450,6 +450,9 @@ int asrc_config_pair(struct asrc_config *config)
 			reg |= 7 << AOCPA;
 		else if ((config->outclk & 0x0f) == OUTCLK_SPDIF_TX)
 			reg |= 6 << AOCPA;
+		else if (((config->outclk & 0x0f) == OUTCLK_ASRCK1_CLK) &&
+				((config->inclk & 0x0f) == INCLK_NONE))
+			reg |= 5 << AOCPA;
 		else if ((config->outclk & 0x0f) == OUTCLK_ASRCK1_CLK) {
 			tmp =
 			    asrc_get_asrck_clock_divider(config->
@@ -492,6 +495,9 @@ int asrc_config_pair(struct asrc_config *config)
 			reg |= 7 << AOCPB;
 		else if ((config->outclk & 0x0f) == OUTCLK_SPDIF_TX)
 			reg |= 6 << AOCPB;
+		else if (((config->outclk & 0x0f) == OUTCLK_ASRCK1_CLK) &&
+				((config->inclk & 0x0f) == INCLK_NONE))
+			reg |= 5 << AOCPB;
 		else if ((config->outclk & 0x0f) == OUTCLK_ASRCK1_CLK) {
 			tmp =
 			    asrc_get_asrck_clock_divider(config->
@@ -534,6 +540,9 @@ int asrc_config_pair(struct asrc_config *config)
 			reg |= 7 << AOCPC;
 		else if ((config->outclk & 0x0f) == OUTCLK_SPDIF_TX)
 			reg |= 6 << AOCPC;
+		else if (((config->outclk & 0x0f) == OUTCLK_ASRCK1_CLK) &&
+				((config->inclk & 0x0f) == INCLK_NONE))
+			reg |= 5 << AOCPC;
 		else if ((config->outclk & 0x0f) == OUTCLK_ASRCK1_CLK) {
 			tmp =
 			    asrc_get_asrck_clock_divider(config->
-- 
1.7.7.4

